exec/plugin-commands-script-runners/src/dlx.ts

```diff
@@ -208,19 +223,24 @@ function createDlxCommandCacheDir (
   opts: {
     registries: Record<string, string>
     cacheDir: string
+    allowBuild: string[]
   }
 ): string {
   const dlxCacheDir = path.resolve(opts.cacheDir, 'dlx')
-  const cacheKey = createCacheKey(pkgs, opts.registries)
+  const cacheKey = createCacheKey(pkgs, opts.registries, opts.allowBuild)
   const cachePath = path.join(dlxCacheDir, cacheKey)
   fs.mkdirSync(cachePath, { recursive: true })
   return cachePath
 }
 
-export function createCacheKey (pkgs: string[], registries: Record<string, string>): string {
+export function createCacheKey (pkgs: string[], registries: Record<string, string>, allowBuild?: string[]): string {
   const sortedPkgs = [...pkgs].sort((a, b) => a.localeCompare(b))
   const sortedRegistries = Object.entries(registries).sort(([k1], [k2]) => k1.localeCompare(k2))
-  const hashStr = JSON.stringify([sortedPkgs, sortedRegistries])
+  const args: unknown[] = [sortedPkgs, sortedRegistries]
+  if (allowBuild?.length) {
+    args.push({ allowBuild: allowBuild.sort(([k1], [k2]) => k1.localeCompare(k2)) })
```

Comment:
This looks wrong. Why are we comparing only the first characters?

```suggestion
    args.push({ allowBuild: allowBuild.sort((k1, k2) => k1.localeCompare(k2)) })
```

Reply:
yes, looks like a mistake. I have copied it from where the registries are sorted